pyiso/ieso.py

```diff
@@ -0,0 +1,649 @@
+                                            parser_format=IESOClient.PARSER_FORMATS.generation)
+        elif self.options.get('start_at', None) and self.options.get('end_at', None):
+            if self.options.get('current_day', False) and not self.options.get('historical', False):
+                range_start = max(self.options['start_at'], gen_out_cap_handler.earliest_available_datetime())
+                range_end = min(self.options['end_at'], gen_out_cap_handler.latest_available_datetime())
+                self._get_report_range(result_ts=generation_ts, report_handler=gen_out_cap_handler,
+                                       parser_format=IESOClient.PARSER_FORMATS.generation, range_start=range_start,
+                                       range_end=range_end)
+            elif self.options.get('yesterday', False):
+                range_start = max(self.options['start_at'], gen_out_cap_handler.earliest_available_datetime())
+                range_end = min(self.options['end_at'], gen_out_cap_handler.latest_available_datetime())
+                self._get_report_range(result_ts=generation_ts, report_handler=gen_out_cap_handler,
+                                       parser_format=IESOClient.PARSER_FORMATS.generation, range_start=range_start,
+                                       range_end=range_end)
+            elif self.options.get('historical', False):
+                self.timeout_seconds = 60  # These reports can get rather large ~7MB for a full year.
+                # For long time ranges at least one day in the past, it is more efficient to request the  Generator
+                # Output by Fuel Type Hourly Report rather than the detailed Generator Output and Capability Report.
+                range_start = max(self.options['start_at'], gen_out_by_fuel_handler.earliest_available_datetime())
+                range_end = min(self.options['end_at'], gen_out_by_fuel_handler.latest_available_datetime())
+                self._get_report_range(result_ts=generation_ts, report_handler=gen_out_by_fuel_handler,
+                                       parser_format=IESOClient.PARSER_FORMATS.generation, range_start=range_start,
+                                       range_end=range_end)
+
+            if self.options.get('forecast', False):
+                range_start = max(self.options['start_at'], self.local_now())
+                range_end = min(self.options['end_at'], adequacy_handler.latest_available_datetime())
+                self._get_report_range(result_ts=generation_ts, report_handler=adequacy_handler,
+                                       parser_format=IESOClient.PARSER_FORMATS.generation, range_start=range_start,
+                                       range_end=range_end)
+        else:
+            LOGGER.warn('No valid options were supplied.')
+        return generation_ts
+
+    def get_load(self, latest=False, yesterday=False, start_at=None, end_at=None, **kwargs):
```

Comment:
The IESO defines "load" to be electricity demand + losses. Is this also the definition which WattTime uses?